### ISSUE TITLE
NameError: global name '_win_crypto' is not defined

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#111]
I get this error on a Linux system (Ubuntu 12).  Started happening with 2.0.  I have also tried 2.0.1.  Here's the stack trace:

```
Traceback (most recent call last):    

  File "main.py", line 420, in <module>
   app.run()    
File "main.py", line 264, in run
    commands.password(args)    
  File "/home/foo/src/pypixapp/pixcli/pixclicommands.py", line 608, in password
    keyring.set_password(keyring_service, user, password)   
  File "/usr/local/lib/python2.7/dist-packages/keyring-2.0.1-py2.7.egg/keyring/core.py", line 42, in set_password
    _keyring_backend.set_password(service_name, username, password)    
  File "/usr/local/lib/python2.7/dist-packages/keyring-2.0.1-py2.7.egg/keyring/backends/file.py", line 85, in set_password
    password_encrypted = self.encrypt(password.encode('utf-8'))    
  File "/usr/local/lib/python2.7/dist-packages/keyring-2.0.1-py2.7.egg/keyring/backends/Windows.py", line 49, in encrypt
    return _win_crypto.encrypt(password)    

NameError: global name '_win_crypto' is not defined
```

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/111
- Originally reported by: [mgoblue80](http://bitbucket.org/mgoblue80)
- Originally created at: 2013-08-26T20:10:27.336

**Comment by jaraco** [comment]
Ugh. That's horrible. I wonder how you ended up in the Windows backend. I'll try to replicate the issue and get a fix out soon.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
I ran into this a while ago. I think the problem is that one of the Windows backends, EncryptedKeyring has a fixed priority of .8 regardless of whether or not it's supported.

This priority is higher than the file based keyrings. If, for whatever reason, you can't use any platform specific keyring in *NIX then the Windows keyring will be used first.

[See here](https://bitbucket.org/kang/python-keyring-lib/src/3827c087ed2710a912a4e202c49e3639472fbcbd/keyring/backends/Windows.py?at=default#cl-38)

---

Original comment by: Dan Soeder

**Comment by jaraco** [comment]
Thanks for the tip. That'll save me the time tracking it down. Sorry for the inconvenience. This will be fixed shortly.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Windows backend shouldn't be viable on non-Windows systems. Fixes #111.

→ <<cset a29b8da4ff26>>

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Thanks for the super fast fix for this one.

---

Original comment by: [mgoblue80](http://bitbucket.org/mgoblue80)
